File: cpp/base/src/main/vsdk/toolkit/environment/geometry/element/Vertex2D.cpp
#include "vsdk/toolkit/environment/geometry/element/Vertex2D.h"
#include "java/lang/String.h"
#include "vsdk/toolkit/common/VSDK.h"
#include "java/lang/String.h"

Vertex2D::Vertex2D(double inX, double inY) : x(inX), y(inY), color() {}
Vertex2D::Vertex2D(double inX, double inY, double r, double g, double b) : x(inX), y(inY), color(r, g, b) {}

java::String Vertex2D::toString() const
{
    return "<" + VSDK::formatDouble(x) + ", " + VSDK::formatDouble(y) + ">";
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/element/Vertex2D.ts
import { FundamentalEntity } from "../../../common/FundamentalEntity.js";
import { VSDK } from "../../../common/VSDK.js";
import { ColorRgb } from "../../../common/color/ColorRgb.js";
export class Vertex2D extends FundamentalEntity {
    public color: ColorRgb;
    public constructor(
        public x: number,
        public y: number,
        r?: number,
        g?: number,
        b?: number,
    ) {
        super();
        this.color = r === undefined ? new ColorRgb() : new ColorRgb(r, g!, b!);
    }
    public override toString() {
        return `<${VSDK.formatDouble(this.x)}, ${VSDK.formatDouble(this.y)}>`;
    }
}
